numpy/core/src/multiarray/arraytypes.c.src

```diff
@@ -3710,6 +3730,65 @@ static void
     npy_intp i;
     @type@ max_val = 0, min_val = 0;
 
+#if HAVE_ATTRIBUTE_TARGET_AVX2 && @type@==npy_int && NPY_BITSOF_@type@ == 32 
+    int unaligned = ni%8;
+    __m128i vec_max,vec_min;
+    __m256i vec_max_256,vec_min_256,vec_array;
+
+    if (max != NULL) {
+       max_val = *max;
+       vec_max = _mm_set1_epi32(max_val);
+       vec_max_256 = _mm256_broadcastd_epi32(vec_max);
+    }
+    if (min != NULL) {
+        min_val = *min;
+        vec_min = _mm_set1_epi32(min_val);
+        vec_min_256 = _mm256_broadcastd_epi32(vec_min);
+	
+    }
+    if (max == NULL) {
+	for(i=0;i<ni/8;i++){
+           vec_array = _mm256_loadu_si256(in+(i*8));
+           vec_array = _mm256_max_epi32(vec_array,vec_min_256);
+           _mm256_storeu_si256(out+(i*8),vec_array);
+       }
+       for(npy_int j=0;j<unaligned;j++){
+             if (in[i*8+j]<min_val) {
+                out[i*8+j]   = min_val;
+            }
+	}
+    	
+    }
+    else if (min == NULL) {
+	for(i=0;i<ni/8;i++){
+       vec_array = _mm256_loadu_si256(in+(i*8));
+       vec_array = _mm256_min_epi32(vec_array,vec_max_256);
```

Comment:
Would be great if this could also use `_mm256_min_epi26` and `_mm256_min_epi8` for other sizes of input